infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -581,5 +626,29 @@ public boolean isConnectionReceiver() {
     public void notifyLeadershipChange() {
         incomingMsgHandler.leadershipChanged();
     }
+
+    public void sendOutSessionToSinkSide(LogReplication.ReplicationSubscriber subscriber) {
+        Set<LogReplicationSession> sessionForSinkSide = this.createOutgoingSessionsBySubscriber(subscriber);
+        for (LogReplicationSession session : sessionForSinkSide) {
+            LogReplication.LogReplicationSinkSessionInitializationMsg msg = LogReplication.LogReplicationSinkSessionInitializationMsg.newBuilder().setSession(session).build();
+            CorfuMessage.RequestPayloadMsg payload = CorfuMessage.RequestPayloadMsg.newBuilder().setLrSinkSessionInitialization(msg).build();
+
+            updateRouterWithNewSessions();
+            createSourceFSMs();
+            logNewlyAddedSessionInfo();
+            if (topology.getRemoteClusterEndpoints().containsKey(session.getSourceClusterId())) {
```

Comment:
```suggestion
            if (router.isConnectionStarterForSession(session)) {
                if (outgoingSessions.contains(session)) {
                    // initiate connection to remote Sink cluster
                    router.connect(topology.getAllClustersInTopology().get(session.getSinkClusterId()), session);
                } else {
                    // initiate connection to remote Source cluster
                    router.connect(topology.getAllClustersInTopology().get(session.getSourceClusterId()), session);
                }
            }
```

Reply:
Also, to avoid code duplication, the above snippet can be put into a new method, say `connect(LogReplicationSession session)`.

Then the code from L563 will change to call the new method as well